pandas/core/indexes/base.py

```diff
@@ -4648,19 +4649,17 @@ def get_value(self, series, key):
                 return libindex.get_value_at(s, key)
             except IndexError:
                 raise
-            except TypeError:
-                # generator/iterator-like
-                if is_iterator(key):
```

Comment:
do we just have no tests which hit this?

Reply:
this should be caught by the `not is_scalar(key)` check that we recently moved to the beginning of this method